sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -17,40 +17,256 @@ namespace Azure.AI.OpenAI
+        public virtual Response<Completions> GetCompletions(CompletionsOptions completionsOptions, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(completionsOptions, nameof(completionsOptions));
+
+            RequestContext context = FromCancellationToken(cancellationToken);
+            Response response = GetCompletions(completionsOptions.ToRequestContent(), context);
+            return Response.FromValue(Completions.FromResponse(response), response);
+        }
+
+        /// <summary> Return the completions for a given prompt. </summary>
+        /// <param name="content"> The content to send as the body of the request. Details of the request body schema are in the Remarks section below. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. Details of the response body schema are in the Remarks section below. </returns>
+        /// <include file="../Generated/Docs/OpenAIClient.xml" path="doc/members/member[@name='GetCompletionsAsync(String,RequestContent,RequestContext)']/*" />
+        public virtual async Task<Response> GetCompletionsAsync(RequestContent content, RequestContext context = null)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+
+            using var scope = ClientDiagnostics.CreateScope("OpenAIClient.GetCompletions");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateGetCompletionsRequest(DeploymentId, content, context);
+                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Return the completions for a given prompt. </summary>
+        /// <param name="content"> The content to send as the body of the request. Details of the request body schema are in the Remarks section below. </param>
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. Details of the response body schema are in the Remarks section below. </returns>
+        /// <include file="../Generated/Docs/OpenAIClient.xml" path="doc/members/member[@name='GetCompletions(String,RequestContent,RequestContext)']/*" />
+        public virtual Response GetCompletions(RequestContent content, RequestContext context = null)
```

Comment:
I don't believe these are new (and thus it's yet another "later" thing), but when/why would a customer use these overloads with manual Response, RequestContent and RequestContext? It's a question for the larger audience, I'd imagine, but it feel strange to me that we have this capability exposed as a peer/sibling to the typed ones.

Reply:
One the underlying typespec gets updated to remove deploymentId from every completions/embeddings method, this method will be autogenerated again as a public virtual. @jpalvarezl Is this just the way the emitter works, making this middle layer method public?